lib/active_resource/associations.rb

```diff
@@ -122,7 +122,12 @@ def defines_belongs_to_finder_method(reflection)
     method_name = reflection.name
     ivar_name = :"@#{method_name}"
 
-    if method_defined?(method_name)
+    defined_methods = self.methods
```

Comment:
I think we should just keep this as is. If people want to define an association called post, they can just not include `CustomMethods` since it will be overridden anyway.